src/IPFSNode.ts

```diff
@@ -68,26 +67,30 @@ class IPFSNode {
 
   private handlePubsubMessage = (msg: IPFS.PubsubMessage): void => {
     if (!(msg && msg.from && msg.data)) {
-      logError(`Message is invalid: ${msg}`);
+      if (typeof msg == 'string') {
+        log('Message is invalid: %s', msg);
+      } else {
+        log('Message is invalid: %j', msg);
```

Comment:
This is very nice, but doesn't this just shift the data processing _(ie: `toString`)_ on `debug`'s shoulders ?

Isn't that' kind of the same in terms of memory usage ?

Reply:
No, because if `debug` is smart this won't be processed at all if debug isn't enabled (I'm assuming it is).

Reply:
That's why they use this kind of string interpolation as opposed to template strings (which are always evaluated)